src/base/utils/fs.cpp

```diff
@@ -518,3 +518,10 @@ QString Utils::Fs::cacheLocation()
         locationDir.mkpath(locationDir.absolutePath());
     return location;
 }
+
+QString Utils::Fs::tempPath()
+{
+    static const QString path = QDir::tempPath() + "/.qBittorrent/";
+    QDir().mkdir(path);
```

Comment:
> This may very well happen with the current approach too. A lock file, maybe, if you want to take care of this situation, would be a better solution?

Sadly, Qt4 doesn't have QLockFile, yet just holding a file open in the temp folder can do the same trick.

Besides, the temp folder can be deleted implies that no object is holding a file in it, i.e. the folder is probably safe to be deleted.